<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="port" default="/dev/ttyACM0"/>
  <arg name="time_out" default="0.5"/>
  <arg name="baudrate" default="115200"/>
  <arg name="imu_topic" default="imu"/>
  <arg name="frame_id" default="imu_link"/>
  <arg name="debug" default="false"/>
  <arg name="rviz_config" default="$(find-pkg-share icm_20948)/rviz2/imu.rviz"/>
  <!-- Imu node -->
  <!--<node pkg="icm_20948" exec="imu_node" output="screen">
    <param name="port" value="$(var port)"/>
    <param name="time_out" value="$(var time_out)"/>
    <param name="baudrate" value="$(var baudrate)"/>
    <param name="imu_topic" value="$(var imu_topic)"/>
    <param name="frame_id" value="$(var frame_id)"/>
  </node>-->
  <group if="$(var debug)">
    <!-- Rviz -->
    <node pkg="rviz2" exec="rviz2" args="-d $(var rviz_config)"/>
    <!-- rqt_plot -->
    <node pkg="rqt_plot" exec="rqt_plot" args="/$(var imu_topic)/linear_acceleration/x:y:z"/>
    <node pkg="rqt_plot" exec="rqt_plot" args="/$(var imu_topic)/angular_velocity/x:y:z"/>
    <node pkg="rqt_plot" exec="rqt_plot" args="/$(var imu_topic)/orientation/x:y:z:w"/>
  </group>
</launch>
